packages/dds/tree/src/util/utils.ts

```diff
@@ -563,3 +563,49 @@ export function defineLazyCachedProperty<
 	});
 	return obj as typeof obj & { [P in K]: V };
 }
+
+/**
+ * Copies a given property from one object to another if and only if the property is present.
```

Comment:
Since JS distinguishes between not present vs present but undefined, we should be careful about the phrasing here: "if and only if the property is present" implies that a present but undefined property will be copied over, but that's not what the implementation does.

Incidentally, it's not very clear to which policy is better.

Reply:
Maybe we can start with the currently implemented policy and add an option later if need be.

Reply:
```suggestion
 * Copies a given property from one object to another if and only if the property value is different from undefined.
 * Note that this means a property `foo` will not be copied from a source object `{ foo: undefined }`.
```

Reply:
Good thoughts. I had similar thoughts regarding future extensibility. I clarified the docs so that terminology ("defined"/"present") is consistent throughout, and spelled out very clearly in the remarks section.

I think the current policy, which does a kind of "normalization" by always preferring the most minimal way to represent an undefined property (that is, by not having it be there at all), is probably what we'll want most of the time. I know we've wanted this behavior a lot, historically, but I can't think of a single time we've ever _wanted_ to add a property with a value of "undefined" rather than simply not having the property there. But you never know.